### PR TITLE
Enhance trace_dump

### PR DESCRIPTION
**Pull request by Gary-Hobson**
## Summary
1: Enhance Android formatted output of NOTE_STRING
    atrace supports many prefixes, not just starting with B/E, such as: ```tracing_mark_write: I|27|10:TSTATE_TASK_PENDING```
2: add INSTRUMENTATION_CSECTION/PREEMPTION support

### CODE REVIEW

File: system/trace/trace_dump.c [hunk 1]
```diff
@@ -616,6 +632,34 @@ static int trace_dump_one(trace_dump_t type, FAR FILE *out, FAR uint8_t *p,
...
 #endif
 
+#ifdef CONFIG_SCHED_INSTRUMENTATION_CSECTION
+      case NOTE_CSECTION_ENTER:
+      case NOTE_CSECTION_LEAVE:
+        {
+          struct note_csection_s *ncs;
+          ncs = (FAR struct note_csection_s *)p;
+          trace_dump_header(out, &ncs->ncs_cmn, ctx);
+          fprintf(out, "tracing_mark_write: %c|%d|critical_section\n",
```
Comment (xiaoxiang781216) [inline system/trace/trace_dump.c:642]: should we output the linux kernel format too?

File: system/trace/trace_dump.c [hunk 2]
```diff
@@ -616,6 +632,34 @@ static int trace_dump_one(trace_dump_t type, FAR FILE *out, FAR uint8_t *p,
...
+#ifdef CONFIG_SCHED_INSTRUMENTATION_PREEMPTION
+      case NOTE_PREEMPT_LOCK:
+      case NOTE_PREEMPT_UNLOCK:
+        {
+          struct note_preempt_s *npr;
+          int16_t count;
+          npr = (FAR struct note_preempt_s *)p;
+          trace_dump_unflatten(&count, npr->npr_count, sizeof(count));
+          trace_dump_header(out, &npr->npr_cmn, ctx);
+          fprintf(out, "tracing_mark_write: %c|%d|sched_lock:%d\n",
```
Comment (xiaoxiang781216) [inline system/trace/trace_dump.c:657]: should we output the linux kernel format too?

File: system/trace/trace_dump.c (hunk 1 above)
Comment (Gary-Hobson) [reply system/trace/trace_dump.c:642]: This is compatible with the linux kernel format.
In the linux kernel, tracing_mark_write is to mark some user data, adding this content will not affect the analysis.
At the same time, I added tracing_mark_write to both dump_string and dump_binary for consistency

**Comment by acassis** [comment]
@Gary-Hobson please include this summary info to the commit message

**Pull request closed without merging**